launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -279,7 +293,63 @@ QString FirstLaunchView::getHeroesInstallDir()
 	return QString{};
 }
 
-void FirstLaunchView::copyHeroesData(const QString & path)
+void FirstLaunchView::extractGogData()
+{
+#ifndef VCMI_NO_INNOEXTRACT
+	QString filterExe = tr("GOG executable") + " (*.exe)";
+	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+	if(fileExe.isEmpty())
+		return;
+
+	QString filterBin = tr("GOG bin file") + " (*.bin)";
+	QString fileBin = QFileDialog::getOpenFileName(this, tr("Select a GOG data (bin) file..."), QFileInfo(fileExe).absolutePath(), filterBin);
```

Comment:
You could check automatically if bin file with the same name exists and suggest to select it only if not found. But not sure if this would work on mobile platforms.

Reply:
Yes. But I think I would wait for Android and maybe simplify it later, if possible.